src/x_connective.c

```diff
@@ -288,13 +288,45 @@ static void send_setup(void)
 /* -------------------- receive ------------------------------ */
 
 static t_class *receive_class;
+static t_class *receive_proxy_class;
+
+typedef struct _receive_proxy
+{
+    t_pd            l_pd;
+    t_inlet        *p_inlet;
+    struct _receive  *p_owner;
+}t_receive_proxy;
 
 typedef struct _receive
 {
     t_object x_obj;
+    t_receive_proxy  pxy; // proxy inlet
     t_symbol *x_sym;
 } t_receive;
 
+static void receive_proxy_init(t_receive_proxy * p, t_receive *x){
+    p->l_pd = receive_proxy_class;
+    p->p_owner = (void *) x;
+}
+
+static void *receive_proxy_new(void)
+{
+    t_receive_proxy *p = (t_receive_proxy *)pd_new(receive_proxy_class);
+    return p;
+}
+
+static void receive_proxy_symbol(t_receive_proxy *p, t_symbol* s){
+    t_receive *x = p->p_owner;
+    pd_unbind(&x->x_obj.ob_pd, x->x_sym);
+    pd_bind(&x->x_obj.ob_pd, x->x_sym = s);
+}
+
+static void receive_proxy_setup(void)
+{
+    receive_proxy_class = (t_class *)class_new(gensym("receive_proxy"), (t_newmethod)receive_proxy_new, 0, sizeof(t_receive_proxy), 0, 0);
```

Comment:
better: 
`receive_proxy_class = (t_class *)class_new(gensym("receive proxy"), 0, 0, sizeof(t_receive_proxy), 0, 0);`

we don't want users to be able to create instances of the proxy class, so we name it "receive proxy" - which is more or less impossible to type - and get rid of the new method - which we don't need anyway.